tests/test_metrics.py

```diff
@@ -0,0 +1,132 @@
+from lsst.ap.pipe.metrics import ApFakesCompletenessMetricTask, ApFakesCompletenessMetricConfig
+
+
+class TestApCompletenessTask(MetricTaskTestCase):
+
+    @classmethod
+    def makeTask(cls, magMin=20, magMax=30):
+        """Make the task and allow for modification of the config min and max.
+
+        Parameters
+        ----------
+        magMin : min magnitude, `float`
+            Minimum magnitude
+        magMax : min magnitude, `float`
+            Maximum magnitude
+        """
+        config = ApFakesCompletenessMetricConfig()
+        config.magMin = magMin
+        config.magMax = magMax
+
+        return ApFakesCompletenessMetricTask(config=config)
+
+    def setUp(self):
+        super().setUp()
+
+        simpleMapConfig = skyMap.discreteSkyMap.DiscreteSkyMapConfig()
+        simpleMapConfig.raList = [45]
+        simpleMapConfig.decList = [45]
+        simpleMapConfig.radiusList = [0.1]
+
+        self.simpleMap = skyMap.DiscreteSkyMap(simpleMapConfig)
+        self.tractId = 0
+        bCircle = self.simpleMap.generateTract(self.tractId).getInnerSkyPolygon().getBoundingCircle()
+        self.targetSources = 1000
+        self.sourceDensity = (self.targetSources
+                              / (bCircle.getArea() * (180 / np.pi) ** 2))
+
+        fakesConfig = CreateRandomApFakesConfig()
+        fakesConfig.fraction = 0.0
+        fakesConfig.fakeDensity = self.sourceDensity
+        fakesTask = CreateRandomApFakesTask(config=fakesConfig)
+        fakeCat = fakesTask.run(self.tractId, self.simpleMap).fakeCat
+
+        self.band = 'g'
+        magCut = 25
+        magMask = (fakeCat[f"{fakesConfig.magVar}" % self.band] < magCut)
+        self.expectedAllMatched = magMask.sum()
+        ids = np.where(magMask, np.arange(1, len(fakeCat) + 1, dtype=int), 0)
+        self.fakeCat = fakeCat.assign(diaObjectId=ids,
+                                      filterName=["g"] * len(fakeCat),
+                                      diaSourceId=ids)
+
+    def testValid(self):
+        """Test the run method.
+        """
+        metricComplete = self.makeTask(20, 30)
+        result = metricComplete.run(self.fakeCat, self.band)
+        lsst.pipe.base.testUtils.assertValidOutput(metricComplete, result)
+
+        meas = result.measurement
+        self.assertEqual(meas.metric_name, Name(metric="ap_pipe.apFakesCompletenessMag20t30"))
+        self.assertEqual(
+            meas.quantity,
+            self.expectedAllMatched / self.targetSources * u.dimensionless_unscaled)
+
+    def testMissingData(self):
+        """Test the run method with no data.
+        """
+        result = self.task.run(None, None)
+        lsst.pipe.base.testUtils.assertValidOutput(self.task, result)
+        meas = result.measurement
+        self.assertIsNone(meas)
+
+    def testValidEmpty(self):
+        """Test the run method with a valid but zero result.
+        """
+        metricComplete = self.makeTask(25, 30)
```

Comment:
I suggest making this
```suggestion
        metricComplete = self.makeTask(self.magCut, self.magCut + 5)
```
(with appropriate changes to `setUp()`). That way the test will be valid if the setup changes later.